pyp2rpm/virtualenv.py

```diff
@@ -0,0 +1,159 @@
+import os
+import logging
+import locale
+from subprocess import Popen, PIPE, DEVNULL
+
+from pyp2rpm.settings import VENV_INTERPRETER
+from pyp2rpm.exceptions import VirtualenvFailException
+
+logger = logging.getLogger(__name__)
+
+PYTHON_VERSION = VENV_INTERPRETER.split('/')[-1]
+
+def site_packages_filter(site_packages_list):
+    '''Removes wheel .dist-info files'''
+    return {x for x in site_packages_list if not x.split('.')[-1] == 'dist-info'}
+
+def deps_package_filter(deps_list, package):
+    '''
+    Removes package name from all installed packages to
+    get dependencies
+    '''
+    return [x for x in deps_list if not x.split('==')[0].lower() == package.lower()]
+
+def deps_wheel_filter(deps_list):
+    '''
+    Removes wheel package from list of installed packages
+    '''
+    return [x for x in deps_list if not x.split('==')[0] == 'wheel']
+
+
+class DirsContent(object):
+    '''
+    Object to store and compare directory content before and
+    after instalation of package.
+    '''
+    def __init__(self, bindir=set(), lib_sitepackages=set()):
+        self.bindir = bindir
+        self.lib_sitepackages = lib_sitepackages
+
+    def fill(self, path):
+        '''
+        Scans content of directories
+        '''
+        self.bindir = set(os.listdir(path + 'bin/'))
+        self.lib_sitepackages = set(os.listdir(path + 'lib/' + PYTHON_VERSION + '/site-packages/'))
+
+    def __sub__(self, other):
+        '''
+        Makes differance of DirsContents objects attributes
+        '''
+        result = DirsContent(
+            self.bindir - other.bindir,
+            self.lib_sitepackages - other.lib_sitepackages)
+        return result
+
+
+class VirtualEnv(object):
+
+    def __init__(self, name, temp_dir, name_convertor):
+        self.name = name
+        self.temp_dir = temp_dir
+        self.name_convertor = name_convertor
+        self.create_virtualenv()
+        self.dirs_before_install = DirsContent()
+        self.dirs_after_install = DirsContent()
+        self.dirs_before_install.fill(self.temp_dir + '/venv/')
+        self.data = {}
+
+    def create_virtualenv(self):
+        '''
+        Creates new virtualenv in temp_dir
+        '''
+        proc = Popen(['virtualenv', '-p', VENV_INTERPRETER, self.temp_dir + '/venv/'],
+                     stdout=DEVNULL, stderr=PIPE)
```

Comment:
Have you considered using the virtualenv-api package? It seems it would be a good fit for this usecase.

[0] https://pypi.python.org/pypi/virtualenv-api/